vivarium/library/dict_utils.py

```diff
@@ -314,6 +314,22 @@ def make_path_dict(embedded_dict):
     return path_dict
 
 
+def apply_func_to_leaves(root: Any, func: Callable[[Any], None]) -> None:
```

Comment:
Love a good dictionary operation, and parametrizing functions (!) Solid addition to `dict_utils` (which could maybe be its own library if it grows large enough.... with a better name).